### ISSUE TITLE
Restore deephaven-learn python tooling

### ISSUE THREAD

**Issue by niloc132** [deephaven/deephaven-core#1133]
CC @chipkent @jmao-denver @jjbrosnan 

Unfortunately, https://github.com/deephaven/deephaven-core/pull/974 had to be reverted (in https://github.com/deephaven/deephaven-core/pull/1096) as the build broke after it merged. Internal slack discussion found that other tests were failing, this ended up being due to #1067 having merged earlier, but not been merged into the #974 branch, so the failure was not detected as part of the pull request.

A two step process is proposed to fix this:

- [ ] Since the deephaven-learn module itself does not require tensorflow, pytorch, scikit, it can be merged on its own in one pull request, so that this tooling is available to integrate the engine's features in whatever "machine learning" toolkit desired by the user. When this is complete, any user can either modify the base images, or use some other tool we hadn't considered
- [ ] Separately, we need to understand the version constraints we are under to make numpy (which we already assume to be installed and work in the engine) and tensorflow/pytorch/scikit/etc get along. The error we encountered here unfortunately was not managed by the package manager pip, but can only be discovered at runtime, so any testing must be done in a running container, probably by adding actual test cases for each tool we support, and running the permutations of installed packages. Finally, size of the full docker image must be considered - tensorflow alone is something like 500mb, which is a significant size increase to our base image, so we almost certainly want to either support several separate images, or allow runtime installation of dependencies.

**Comment by chipkent** [comment]
Updating status.

1) Learn library without dependencies has gone in.  #1140 
2) Learn library dependencies have been blocked by base+ image work.  @niloc132, @devinrsmith, or @jjbrosnan can comment on status.
3) We still need to run testing on the base+ images.

I'm going to assign to @devinrsmith and @niloc132, since their work has been blocking.
